GeneratedRegexPatterns/MediumConfidenceSecurityModels.json

```diff
@@ -1,51 +1,51 @@
 [
   {
-    "Pattern": "(?i)\\.servicebus\\.windows.+[^0-9a-z\\/+](?P<refine>[0-9a-z\\/+]{43}=)(?:[^=]|$)",
+    "DetectionMetadata": "ObsoleteFormat, HighEntropy, MediumConfidence",
     "Id": "SEC101/105",
     "Name": "AzureMessageLegacyCredentials",
+    "Pattern": "(?i)\\.servicebus\\.windows.+[^0-9a-z\\/+](?P<refine>[0-9a-z\\/+]{43}=)(?:[^=]|$)",
     "Signatures": [
       ".servicebus"
-    ],
-    "DetectionMetadata": "ObsoleteFormat, HighEntropy, MediumConfidence"
+    ]
   },
   {
-    "Pattern": "(?:^|[^0-9A-Za-z-_.])e[0-9A-Za-z-_=]{23,}\\.e[0-9A-Za-z-_=]{23,}\\.[0-9A-Za-z-_=]{24,}(?:[^0-9A-Za-z-_]|$)",
+    "DetectionMetadata": "HighEntropy, MediumConfidence",
     "Id": "SEC101/528",
     "Name": "GenericJwt",
+    "Pattern": "(?:^|[^0-9A-Za-z-_.])e[0-9A-Za-z-_=]{23,}\\.e[0-9A-Za-z-_=]{23,}\\.[0-9A-Za-z-_=]{24,}(?:[^0-9A-Za-z-_]|$)",
```

Comment:
>- [](http://example.com/codeflow?start=116&length=1)

unescaped hyphens make me nervous in these range constructs. if someone introduced a character before this hyphen it would change it to a separator between a start and trailing char. if you add an underscore before the hyphen, for example, the hyphen would now be allowed to match.

you didn't create the problem, but can we add a work item to ensure this hyphen and all other hyphens in ranges are escaped?

Reply:
Filed #178